instrumentation/opentelemetry-instrumentation-openai/tests/test_chat_completions.py

```diff
@@ -0,0 +1,162 @@
+    ]
+
+    kwargs = {
+        "model": llm_model_value,
+        "messages": messages_value,
+        "stream": False,
+    }
+
+    response = openai_client.chat.completions.create(**kwargs)
+    spans = exporter.get_finished_spans()
+    chat_completion_span = spans[-1]
+    # assert that the span name is correct
+    assert chat_completion_span.name == f"chat {llm_model_value}"
+
+    attributes = chat_completion_span.attributes
+    operation_name = attributes[GenAIAttributes.GEN_AI_OPERATION_NAME]
+    system = attributes[GenAIAttributes.GEN_AI_SYSTEM]
+    request_model = attributes[GenAIAttributes.GEN_AI_REQUEST_MODEL]
+    response_model = attributes[GenAIAttributes.GEN_AI_RESPONSE_MODEL]
+    response_id = attributes[GenAIAttributes.GEN_AI_RESPONSE_ID]
+    input_tokens = attributes[GenAIAttributes.GEN_AI_USAGE_INPUT_TOKENS]
+    output_tokens = attributes[GenAIAttributes.GEN_AI_USAGE_OUTPUT_TOKENS]
+
+    # assert that the attributes are correct
+    assert (
+        operation_name == GenAIAttributes.GenAiOperationNameValues.CHAT.value
+    )
+    assert system == GenAIAttributes.GenAiSystemValues.OPENAI.value
+    assert request_model == llm_model_value
+    assert response_model == response.model
+    assert response_id == response.id
+    assert input_tokens == response.usage.prompt_tokens
+    assert output_tokens == response.usage.completion_tokens
+
+    events = chat_completion_span.events
+
+    # assert that the prompt and completion events are present
+    prompt_event = list(
+        filter(
+            lambda event: event.name == "gen_ai.content.prompt",
+            events,
+        )
+    )
+    completion_event = list(
+        filter(
+            lambda event: event.name == "gen_ai.content.completion",
+            events,
+        )
+    )
+
+    assert prompt_event
+    assert completion_event
+
+    # assert that the prompt and completion events have the correct attributes
+    assert prompt_event[0].attributes[
+        GenAIAttributes.GEN_AI_PROMPT
+    ] == json.dumps(messages_value)
+
+    assert (
+        json.loads(
+            completion_event[0].attributes[GenAIAttributes.GEN_AI_COMPLETION]
+        )[0]["content"]
+        == response.choices[0].message.content
+    )
+
+
+@pytest.mark.vcr()
+def test_chat_completion_streaming(exporter, openai_client):
+    llm_model_value = "gpt-4"
+    messages_value = [
+        {"role": "user", "content": "Say this is a test three times"}
+    ]
+
+    kwargs = {
+        "model": llm_model_value,
+        "messages": messages_value,
+        "stream": True,
+        "stream_options": {"include_usage": True},
+    }
+
+    response_stream_usage = None
+    response_stream_model = None
+    response_stream_id = None
+    response_stream_result = ""
+    response = openai_client.chat.completions.create(**kwargs)
+    for chunk in response:
+        if chunk.choices:
+            response_stream_result += chunk.choices[0].delta.content or ""
```

Comment:
maybe as an assumption, we have an assertion that there were in fact multiple chunks. This helps prove later assumptions, such as that the event from a stream is a consolidation of the chunks in it.